### DISCUSSION TITLE
Blame in diffs

### DISCUSSION THREAD

**Discussion by StephanHoyer** [community/community#12968] [Pull Requests]
Would be great to see commit messages of changes in the diff/pr view, so I can inspect, why this change was made. This would especially useful for larger pr's with a lot of commits.

**Comment by tbrlpld** [comment]
Yup. Keep wanting this myself. That would be super useful.

**Comment by hlongmore-elation** [comment]
💯 I have a bunch of "why was this done" questions to answer on a PR tonight, that, if they would have taken the suggestion I left on a comment at the top of the diffs to go view the commits instead of the diff, they _maybe_ would have not had to ask, since I explained it all in the commit messages. But now, I have to go find the commit messages, and copy and paste them into the PR, or at least copy and paste links to the commits, or at worst, re-explain everything. I was going to write this as a feature request like so:

```
This is one thing where I think github's interface for PRs could be improved. I included the rationale in
the commit messages. It would be nice if github could have like a question mark that when clicked, would
show the commit message for the particular line.
```

**Comment by 0xTadash1** [comment]
+1

There are feature requests similar to this:

- https://github.com/orgs/community/discussions/31585
- https://github.com/orgs/community/discussions/7465

I think it would be convenient if **inline blame** is displayed when hovering over with the mouse.

### "Inline blame" Example:

Inline blame feature of [gitkraken/vscode-gitlens](https://github.com/gitkraken/vscode-gitlens?tab=readme-ov-file) extension.

> ![img](https://raw.githubusercontent.com/gitkraken/vscode-gitlens/main/images/docs/current-line-blame.png)

**Comment by joelostblom** [comment]
I agree that this would be really helpful. Often when I review the changes made in a PR in the "Changed files" tab, I wonder why a certain line was changed. Currently I would have to toggle to the commit tabs and manually find the commit that introduced the change to see the author's motivation for changing those lines. Since this takes several manual steps, I'm less likely to do it and I also notice that other PR reviewers rarely read commit messages. Commit messages are helpful and it is unfortunate that they are not more readily accessible when viewing the lines changed in a PR on github.com. It would be helpful for reviewers if GitHub could make it so that commit message popped up when hovering over a changed line (or when the line is selected).

The current workaround are to duplicate commit messages as comments which is not ideal, or to go through a PR commit by commit, which can be rather confusing when there are many commits and the lines are changed multiple times.
